src/main/java/com/n/opensource/reddoit/controllers/api/RecipeController.java

```diff
@@ -18,18 +24,49 @@ public class RecipeController {
     private final RecipeService recipeService;
 
     @PostMapping("/create")
-    public ResponseEntity<RecipeDTO> createRecipe(@RequestBody CreateRecipeRequest createRecipeRequest){
+    public ResponseEntity<RecipeDTO> createRecipe(@RequestBody CreateRecipeRequest createRecipeRequest) {
         return ResponseEntity.ok(recipeService.createRecipe(createRecipeRequest));
     }
 
     @GetMapping("/getPost")
-    public ResponseEntity<RecipeDTO> getRecipeById(@RequestParam(name = "id")UUID recipeId){
+    public ResponseEntity<RecipeDTO> getRecipeById(@RequestParam(name = "id") UUID recipeId) {
         return ResponseEntity.ok(recipeService.getRecipeById(recipeId));
     }
+
     @DeleteMapping("/delete")
-    public ResponseEntity<String> deleteRecipe(@RequestParam(name = "id")UUID recipeId){
+    public ResponseEntity<String> deleteRecipe(@RequestParam(name = "id") UUID recipeId) {
         recipeService.deleteRecipe(recipeId);
         String response = "Recipe has been deleted";
         return ResponseEntity.ok(response);
     }
+
+    @PostMapping("/banner/{id}")
+    public ResponseEntity<Recipe> setRecipeBanner(@PathVariable UUID id, @RequestParam("file") MultipartFile file) {
+        try {
+            Recipe recipe = recipeService.setRecipeBanner(id, file);
+            return ResponseEntity.status(HttpStatus.CREATED).body(recipe);
+        } catch (IOException e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
+        }
+    }
+
+    @PostMapping("/foodtags/{id}")
+    public ResponseEntity<Recipe> setRecipeFoodTags(@PathVariable UUID id, @RequestParam SetRecipeFoodTagsRequest setRecipeFoodTagsRequest) {
```

Comment:
Why send the recipe back in the body of the response? If you want to get back the recipe page after setting tags for a recipe, wouldn't it be better to just query it using the getRecipeById endpoint?